### PR TITLE
Add RCR Controller Methods

### LINKED ISSUE #432: RecruitingCampaignRecipient endpoint should not add if email is already in campaign

### PR DESCRIPTION
**Pull request by RossIV**
Adds functioning methods to RCR controller

Closes #432.

### CODE REVIEW

File: app/Http/Controllers/RecruitingCampaignRecipientController.php [hunk 1]
```diff
@@ -27,7 +27,47 @@ public function index()
      */
     public function store(Request $request)
     {
-        return response()->json(['status' => 'error', 'error' => 'not_implemented'], 501);
+        $this->validate($request, [
+            'recruiting_campaign_id' => 'exists:recruiting_campaigns,id|numeric',
+            'addresses' => 'required',
+            'recruiting_visit_id' => 'exists:recruiting_visits,id',
```
[the remark this answers is not in the document]
Comment (RossIV) [reply app/Http/Controllers/RecruitingCampaignRecipientController.php]: That's a valid point that I didn't consider. I'll have to think about how I want to handle that...

File: app/Http/Controllers/RecruitingCampaignRecipientController.php [hunk 2]
```diff
@@ -27,7 +27,48 @@ public function index()
...
+            'address' => 'required_unless:recipients',
+            'recruiting_visit_id' => 'exists:recruiting_visits,id',
+            'user_id' => 'exists:users,id|numeric',
+        ]);
+
+        if ($request->filled('recipients')) {
+            $recipients = $request->input('recipients');
+        } else {
+            $recipients[0]['address'] = $request->input('address');
+            $recipients[0]['recruiting_visit_id'] = $request->input('address');
```
Comment (kberzinch) [inline app/Http/Controllers/RecruitingCampaignRecipientController.php]: Typo?

Comment (kberzinch) [reply app/Http/Controllers/RecruitingCampaignRecipientController.php]: Also should you be setting `recruiting_campaign_id` for this recipient?

Comment (RossIV) [reply app/Http/Controllers/RecruitingCampaignRecipientController.php]: >Typo?
Yes

>Also should you be setting recruiting_campaign_id for this recipient?
Assumption was one campaign per request

**Comment by kberzinch** [comment]
I can't seem to get any of these endpoints to do anything useful in Postman. How would I add a recipient to a campaign, for example?

**Comment by RossIV** [comment]
Theoretically you should be able to:
`POST` `/api/v1/recruiting/campaigns/recipients` with
```
{
  "recruiting_campaign_id": 5,
   "recipients":{
      "0":{
         "email_address":"test@abc.net",
         "recruiting_visit_id":2,
         "user_id":1
      }
   }
}
```
to add a recipient to a campaign.

`GET` `/api/v1/campaigns/recipients` to get all recipients for a campaign

`update()` is for individual RCRs, so different syntax that `store()`

**Comment by kberzinch** [comment]
I get a 405 Method Not Allowed when adding and just the following response when GETting

```javascript
{
    "status": "success",
    "campaign": []
}
```

**Comment by RossIV** [comment]
Sounds like I messed up the routes. Will look after football 🏈.

**Pull request merged**